catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/CatalogMigratableConfig.java

```diff
@@ -0,0 +1,139 @@
+    private int exportThreadCount;
+
+    public CatalogMigratableConfig() {
+        exportPath = Paths.get("migration");
+        this.exportFilePrefix = "catalogExport";
+        this.exportCardsPerFile = 1;
+        this.exportQueryPageSize = 5000;
+        this.exportThreadCount = 8;
+    }
+
+    public Path getExportPath() {
+        return exportPath;
+    }
+
+    public void setExportPath(Path exportPath) {
+        this.exportPath = exportPath;
+    }
+
+    public int getExportCardsPerFile() {
+        return exportCardsPerFile;
+    }
+
+    public void setExportCardsPerFile(int exportCardsPerFile) {
+        if (exportCardsPerFile >= 1 && exportCardsPerFile <= MAX_CARDS_PER_FILE) {
+            if (exportCardsPerFile <= this.exportQueryPageSize) {
+                this.exportCardsPerFile = exportCardsPerFile;
+            } else {
+                LOGGER.error("Cards per file must be less than or equal to query page size");
+                throw new IllegalArgumentException(
+                        "Cards per file must be less than or equal to query page size");
+            }
+        } else {
+            LOGGER.error("Cards per file is invalid; must be between {} and {}",
+                    1,
+                    MAX_CARDS_PER_FILE);
+            throw new IllegalArgumentException(String.format(
+                    "Cards per file is invalid; must be between %d and %d",
+                    1,
+                    MAX_CARDS_PER_FILE));
+        }
+    }
+
+    public int getExportQueryPageSize() {
+        return exportQueryPageSize;
+    }
+
+    public void setExportQueryPageSize(int exportQueryPageSize) {
+        if (exportQueryPageSize >= MIN_QUERY_PAGE_SIZE
+                && exportQueryPageSize <= MAX_QUERY_PAGE_SIZE) {
+            this.exportQueryPageSize = exportQueryPageSize;
+        } else {
+            LOGGER.error("Query page size is invalid; must be between {} and {}",
+                    MIN_QUERY_PAGE_SIZE,
+                    MAX_QUERY_PAGE_SIZE);
+            throw new IllegalArgumentException(String.format(
+                    "Query page size is invalid; must be between %d and %d",
+                    MIN_QUERY_PAGE_SIZE,
+                    MAX_QUERY_PAGE_SIZE));
+        }
+    }
+
+    public int getExportThreadCount() {
+        return exportThreadCount;
+    }
+
+    public void setExportThreadCount(int exportThreadCount) {
+        if (exportThreadCount >= 1 && exportThreadCount <= MAX_THREADS) {
+            this.exportThreadCount = exportThreadCount;
+        } else {
+            LOGGER.error("Thread count is invalid; must be between 1 and {}", MAX_THREADS);
+            throw new IllegalArgumentException(String.format(
+                    "Thread count is invalid; must be between 1 and %s",
+                    MAX_THREADS));
+        }
+    }
+
+    public String getExportFilePrefix() {
+        return exportFilePrefix;
+    }
+
+    public void setExportFilePrefix(String prefix) {
```

Comment:
My concern here is that the reason we would provide a setter and do all this validation is because we're considering exposing this on the Admin UI. Which means that administrators _could_ use valid file name characters that would be rejected right now, which is a bad user experience.

It also forces us to write and maintain a bunch of unit and integration tests for a use case that I don't think we really need to support.